src/main/scala/cognite/spark/v1/RelationshipsRelation.scala

```diff
@@ -0,0 +1,191 @@
+        "minEndTime",
+        "maxEndTime",
+        "minConfidence",
+        "maxConfidence",
+        "dataSetId",
+        "labels",
+        "minCreatedTime",
+        "maxCreatedTime",
+        "minLastUpdatedTime",
+        "maxLastUpdatedTime"
+      )
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val shouldGetAllRows = shouldGetAll(pushdownFilterExpression, fieldNames)
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+
+    if (filtersAsMaps.isEmpty || shouldGetAllRows) {
+      Seq(client.relationships.filter(RelationshipsFilter()).map(relationshipToRelationshipReadSchema))
+    } else {
+      val relationshipsFilterSeq = filtersAsMaps.distinct.map(relationshipsFilterFromMap)
+      relationshipsFilterSeq
+        .map { f =>
+          client.relationships.filter(f, limit).map(relationshipToRelationshipReadSchema)
+        }
+    }
+
+  }
+
+  def relationshipsFilterFromMap(m: Map[String, String]): RelationshipsFilter =
+    RelationshipsFilter(
+      sourceExternalIds = getExternalIdSeqFromExternalId(m.get("sourceExternalId")),
+      sourceTypes = getExternalIdSeqFromExternalId(m.get("sourceType")),
+      targetExternalIds = getExternalIdSeqFromExternalId(m.get("targetExternalId")),
+      targetTypes = getExternalIdSeqFromExternalId(m.get("targetType")),
+      dataSetIds = m.get("dataSetId").map(idsFromWrappedArray(_).map(CogniteInternalId)),
+      startTime = timeRangeFromMinAndMax(m.get("minStartTime"), m.get("maxStartTime")),
+      endTime = timeRangeFromMinAndMax(m.get("minEndTime"), m.get("maxEndTime")),
+      labels = m.get("labels").map(externalIdsToContainsAny).getOrElse(None),
+      confidence = confidenceRangeFromLimitStrings(m.get("minConfidence"), m.get("maxConfidence")),
+      lastUpdatedTime = timeRangeFromMinAndMax(m.get("minLastUpdatedTime"), m.get("maxLastUpdatedTime")),
+      createdTime = timeRangeFromMinAndMax(m.get("minCreatedTime"), m.get("maxCreatedTime"))
+    )
+
+  override def insert(rows: Seq[Row]): IO[Unit] = {
+    val relationships =
+      rows.map(fromRow[RelationshipsInsertSchema](_)).map(relationshipInsertSchemaToRelationshipCreate)
+    client.relationships
+      .create(relationships)
+      .flatTap(_ => incMetrics(itemsCreated, relationships.length)) *> IO.unit
+  }
+
+  override def delete(rows: Seq[Row]): IO[Unit] = {
+    val relationshipIds = rows.map(fromRow[RelationshipsDeleteSchema](_)).map(_.externalId)
+    client.relationships
+      .deleteByExternalIds(relationshipIds)
+      .flatTap(_ => incMetrics(itemsDeleted, relationshipIds.length))
+  }
+
+  override def upsert(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Upsert is not supported for relationships")
+
+  override def update(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Update is not supported for relationships")
+
+  def relationshipToRelationshipReadSchema(relationship: Relationship): RelationshipsReadSchema =
```

Comment:
This is only needed for labels, if there is a better way to do this conversion without relationshipToRelationshipReadSchema function, that would be great. This is only needed for labels.

It uses the same type for all except for labels. Labels come as  ```{"labels": [{external_id: "label"}, {"external_id": "label2"}]}``` on Relationship objects. I don't know why API is designed like this since labels only have external ids. It could be simply ```{labels: ["label1", "label2"]}``` and it would make this whole work a one day thing. I wanted to be fast so I tried to implement this without understanding completely. After spending a long time and seeing that will not work , I decided to understand and found this conversion idea. This could be messy when adding assets labels field. Because there we have update/upsert.

Reply:
For now, I pass null to labels on AssetsRelation. It will be a separate task to make it available for assets.

Reply:
df.where("labels in(array('test_label'))").show() or df.where("labels = array('test_label')").show()  is working with this workaround. Also, it is possible to provide a list of labels when creating relationships.

Reply:
So this would have worked with `Relationships` if this https://github.com/cognitedata/cdp-spark-datasource/pull/476 was merged, but it's still a draft and I don't want to mix the two topics into this PR so let's do it like this for now and then change it when/if we get #476 merged.

Reply:
It would be great to have this change. I guess it is not ideal to implement a conversion for all types we have labels (if all types will have it at some point).